dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageCheckRequest.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.opensource.classpath;
+
+import com.google.cloud.tools.opensource.dependencies.Bom;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.List;
+import org.eclipse.aether.artifact.Artifact;
+
+/**
+ * Request to run Linkage Check for {@link #classPath}.
+ *
+ * <p>The check should suppress linkage errors that match {@link #excludedErrors}. When {@link
+ * #reportOnlyReachable}, it should suppress linkage errors on the classes not reachable from {@link
+ * #entryPoints}.
+ */
+public class LinkageCheckRequest {
+
+  private final ImmutableList<ClassPathEntry> classPath;
```

Comment:
Here's what bother some about this. There are two kinds of data here being treated the same. 

The first is the classpath itself. This models the thing in the domain we want to check. Arguably excludedErrors is this kind as well, to the extent we think of it as exisiting or irrelevant errors. These are properties of the specific jar files we're checking.

Then there's entry points and reportOnlyReachable. These are decisions made to parameterize the algorithm we're going to run. They are not intrinsic characteristics of the data that exists outside the program.  We could easily recheck the same classpath with different entry points or different choice of whether to report unreachable errors. 

These two kinds of things don't belong in the same class.